File: Models/Employee.cs
using dotnet_backend_freshmart.Models.Enums;

namespace dotnet_backend_freshmart.Models
{
        public class Employee
        {
            public Guid Id { get; set; }

            public string Code { get; set; } = string.Empty;

            public string Name { get; set; } = string.Empty;

            public string? Phone { get; set; }

            public string? Email { get; set; }

            public EmployeeRole Role { get; set; } = EmployeeRole.Cashier;

            public string? PinHash { get; set; }

            public bool IsActive { get; set; } = true;

            public DateTime? HiredDate { get; set; }

            public string? Notes { get; set; }

            public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

            public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        }
}

File: freshmart/src/types/employee.ts
// ============================================================================
// Types cho module Quản lý Nhân sự (Employees)
// Khớp 100% với DTOs của Backend ASP.NET Core
// ============================================================================

export type EmployeeRoleType = 0 | 1 | 2 | 3 | 'Cashier' | 'StoreManager' | 'WarehouseStaff' | 'Admin';

export interface Employee {
  id: string;
  code: string;
  name: string;
  phone?: string | null;
  email?: string | null;
  role: string;
  isActive: boolean;
  hiredDate?: string | null;
  notes?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface EmployeeFilterParams {
  search?: string;
  role?: string | number;
  isActive?: boolean;
  page?: number;
  limit?: number;
}

export interface CreateEmployeeRequest {
  name: string;
  phone: string;
  email?: string;
  role: EmployeeRoleType;
  pin: string;
  hiredDate?: string;
  notes?: string;
}

export interface UpdateEmployeeRequest {
  name: string;
  phone: string;
  email?: string;
  role: EmployeeRoleType;
  hiredDate?: string;
  notes?: string;
}

export interface ResetPinRequest {
  newPin: string;
}
